lib/azure/armrest/exception.rb

```diff
@@ -52,18 +52,16 @@ def to_s
       end
     end
 
-    # A list of predefined exceptions that we wrap around RestClient exceptions.
+    # Redefine the RestClient exception classes as our own.
+    RestClient::Exceptions::EXCEPTIONS_MAP.values.each do |klass|
+      name = klass.to_s.split("::").last
+      if name != 'Exception'
+        name += 'Exception' unless name.include?('Exception')
+        Object.module_eval("class Azure::Armrest::#{name} < Azure::Armrest::ApiException; end")
```

Comment:
😕 

Can you utilize something like `MyError = Class.new(ErrorImInheritingFrom)` instead of `Object.module_eval`? (The dynamic nature of this has some small caveats you'll need to consider but I think with a little research it's possible)